src/main/java/io/github/hapjava/server/impl/crypto/HAPSetupCodeUtils.java

```diff
@@ -0,0 +1,74 @@
+package io.github.hapjava.server.impl.crypto;
+
+import java.util.Base64;
+import org.bouncycastle.crypto.Digest;
+import org.bouncycastle.crypto.digests.SHA512Digest;
+
+public class HAPSetupCodeUtils {
+  private static final String ALPHA_NUMERIC_STRING = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+
+  public static String randomAlphaNumeric(int count) {
+    StringBuilder builder = new StringBuilder();
+    while (count-- != 0) {
+      int character = (int) (Math.random() * ALPHA_NUMERIC_STRING.length());
+      builder.append(ALPHA_NUMERIC_STRING.charAt(character));
+    }
+    return builder.toString();
+  }
+
+  public static String generateSetupId() {
+    String setupId =
```

Comment:
why aren't you just calling randomAlphaNumeric here?

Reply:
right, i have copy randomAlphaNumeric from somewhere first and then it was looking for me to monstrous for just 4 random characters. 
but you are right, i should either use it and remove it.

Reply:
done